src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.cpp

```diff
@@ -534,13 +534,31 @@ void os::verify_stack_alignment() {
 }
 #endif
 
+static THREAD_LOCAL WXMode _wx_state = WXUnknown;
```

Comment:
All this wx coding inside bsd sources should be guarded with `__APPLE__` out of politeness toward the BSDs.